### ISSUE TITLE
jq-mode

### ISSUE THREAD

**Issue by ljos** [rejeep/jq.el#2]
Hi, I just saw this after starting my own [jq-mode](https://github.com/ljos/jq-mode). Maybe we should merge our efforts?

**Comment by rejeep** [comment]
Sound good! :+1: 

I don't have a lot of time hacking of stuff like this these days, so maybe we should continue on yours!

**Comment by ljos** [comment]
I added you as a collaborator so you can merge in the things from here that you need. I don't completely understand what everything does so I am a bit afraid of messing something up by just moving it over.

**Comment by rejeep** [comment]
The purpose of my jq-mode is to write a jq-expression and get instant feedback in the JSON-buffer you are viewing. Kind of like Emacs's `re-builder`. Your mode more looks like you are building a major mode to edit jq-expressions?

**Comment by ljos** [comment]
Yes. That is true.

**Comment by rejeep** [comment]
Added a small demo to the README, check it out

**Comment by ljos** [comment]
Cool! I think it would make sense to add an `interactive-mode` to my project like that. I can look at how to join the two projects now that I understand what it is that you are trying to do.

**Comment by rejeep** [comment]
Makes sense. Take a look at the tests as well. Should be easier to migrate then as you can just run the tests and make sure it works.

**Comment by ljos** [comment]
Thanks. I will try. (Though right now I will get some dinner...)
